src/Config/MergeConfig.cs

```diff
@@ -0,0 +1,126 @@
+            {
+                JsonElement root1 = originalJsonDoc.RootElement;
+                JsonElement root2 = overridingJsonDoc.RootElement;
+
+                if (root1.ValueKind != JsonValueKind.Array && root1.ValueKind != JsonValueKind.Object)
+                {
+                    throw new InvalidOperationException($"The original JSON document to merge new content into must be a container type. Instead it is {root1.ValueKind}.");
+                }
+
+                if (root1.ValueKind != root2.ValueKind)
+                {
+                    return originalJson;
+                }
+
+                if (root1.ValueKind == JsonValueKind.Array)
+                {
+                    MergeArrays(jsonWriter, root2);
+                }
+                else
+                {
+                    MergeObjects(jsonWriter, root1, root2);
+                }
+            }
+
+            return Encoding.UTF8.GetString(outputBuffer.WrittenSpan);
+        }
+
+        /// <summary>
+        /// This helper methods tries to merge two Json Object recursively.
+        /// </summary>
+        private static void MergeObjects(Utf8JsonWriter jsonWriter, JsonElement root1, JsonElement root2)
+        {
+            jsonWriter.WriteStartObject();
+
+            // If a property exists in both documents, either:
+            // * Merge them, if the value kinds match (e.g. both are objects or arrays) recursively,
+            // * Completely override the value of the first with the one from the second, if the value kind mismatches (e.g. one is object, while the other is an array or string),
+            // * Or favor the value of the first, if the second one is null.
+            foreach (JsonProperty property in root1.EnumerateObject())
+            {
+                string propertyName = property.Name;
+
+                JsonValueKind newValueKind;
+
+                if (root2.TryGetProperty(propertyName, out JsonElement newValue) && (newValueKind = newValue.ValueKind) != JsonValueKind.Null)
+                {
+                    jsonWriter.WritePropertyName(propertyName);
+
+                    JsonElement originalValue = property.Value;
+                    JsonValueKind originalValueKind = originalValue.ValueKind;
+
+                    if (newValueKind == JsonValueKind.Object && originalValueKind == JsonValueKind.Object)
+                    {
+                        MergeObjects(jsonWriter, originalValue, newValue); // Recursive call
+                    }
+                    else if (newValueKind == JsonValueKind.Array && originalValueKind == JsonValueKind.Array)
+                    {
+                        MergeArrays(jsonWriter, newValue);
+                    }
+                    else
+                    {
+                        newValue.WriteTo(jsonWriter);
+                    }
+                }
+                else
+                {
+                    property.WriteTo(jsonWriter);
```

Comment:
>property.WriteTo(jsonWriter); [](http://example.com/codeflow?start=20&length=29)

Can you add more comments? this else case means the override config does not provide a different value? (only for JSON Value Types: Number, True, False, string)

Reply:
yes, no override value available. So just add whats available in the first config. Added the comments.